lightning-invoice/src/payment.rs

```diff
@@ -1364,18 +1549,80 @@ mod tests {
 		invoice_payer.handle_event(&event);
 	}
 
+	#[test]
+	fn inflight_data_trivial_test() {
+		let event_handled = core::cell::RefCell::new(false);
+		let event_handler = |_: &_| { *event_handled.borrow_mut() = true; };
+
+		let payment_preimage = PaymentPreimage([1; 32]);
+		let invoice = invoice(payment_preimage);
+		let payment_hash = Some(PaymentHash(invoice.payment_hash().clone().into_inner()));
+		let final_value_msat = invoice.amount_milli_satoshis().unwrap();
+
+		let payer = TestPayer::new().expect_send(Amount::ForInvoice(final_value_msat));
+		let final_value_msat = invoice.amount_milli_satoshis().unwrap();
+		let route = TestRouter::route_for_value(final_value_msat);
+		let router = TestRouter {};
+		let scorer = RefCell::new(TestScorer::new());
+		let logger = TestLogger::new();
+		let invoice_payer =
+			InvoicePayer::new(&payer, router, &scorer, &logger, event_handler, Retry::Attempts(0));
+
+		let payment_id = invoice_payer.pay_invoice(&invoice).unwrap();
+
+		let inflight_map = invoice_payer.create_inflight_map(&payment_hash.unwrap());
+		// First path check
+		assert_eq!(inflight_map.get(&(0, true)).unwrap().clone(), 94);
+		assert_eq!(inflight_map.get(&(1, false)).unwrap().clone(), 84);
+		assert_eq!(inflight_map.get(&(2, true)).unwrap().clone(), 64);
+
+		// Second path check
+		assert_eq!(inflight_map.get(&(1, true)).unwrap().clone(), 64);
+
+		invoice_payer.handle_event(&Event::PaymentPathSuccessful {
+			payment_id, payment_hash, path: route.paths[0].clone()
+		});
+
+		assert_eq!(inflight_map.get(&(0, true)), None);
+		assert_eq!(inflight_map.get(&(1, false)), None);
+		assert_eq!(inflight_map.get(&(2, true)), None);
+
+		// Second path should still be inflight
+		assert_eq!(inflight_map.get(&(1, true)).unwrap().clone(), 64);
```

Comment:
You'll need to create the in-flight map again before checking these expectations.